jgrapht-core/src/main/java/org/jgrapht/alg/cycle/HawickJamesSimpleCycles.java

```diff
@@ -213,36 +188,16 @@ private void unblock(Integer u)
         for (int wPos = 0; wPos < B[u].size(); wPos++) {
             Integer w = B[u].get(wPos);
 
-            wPos -= removeFromList(B[u], w);
+            wPos -= B[u].size();
+            B[u].removeAll(singletonList(w));
+            wPos -= B[u].size();
```

Comment:
Errr, I meant:

```
            int sizeBeforeRemove = B[u].size();
            B[u].removeAll(singletonList(w));
            wPos -= (sizeBeforeRemove - B[u].size());
```

Reply:
Oh, sorry, thanks for the clarification:) 
I fixed it:)